src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/CloudStackEventHandler.scala

```diff
@@ -6,22 +6,46 @@ import com.bwsw.cloudstack.vault.server.common.traits.EventHandler
 import com.bwsw.cloudstack.vault.server.controllers.CSVaultController
 import org.slf4j.LoggerFactory
 
+import scala.concurrent.{ExecutionContext, Future}
+import CloudStackEvent.Action._
+
 /**
   * Created by medvedev_vv on 02.08.17.
   */
-class CloudStackEventHandler(controller: CSVaultController) extends EventHandler {
-  private val jsonSerializer = new JsonSerializer()
+class CloudStackEventHandler(controller: CSVaultController)
+                            (implicit executionContext: ExecutionContext) extends EventHandler[CloudStackEvent] {
+  private val jsonSerializer = new JsonSerializer(true)
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  def handleEvent(value: String): Unit = {
-    import CloudStackEvent.Action._
-    val event = jsonSerializer.deserialize[CloudStackEvent](value)
-    event.action match {
-      case VMCreate => logger.debug("VMCreate")
-      case VMDelete => logger.debug("VMDelete")
-      case AccountCreate => logger.debug("AccountCreate")
-      case AccountDelete => logger.debug("AccountDelete")
-      case UserCreate => logger.debug("UserCreate")
+  @Override
+  def handleEventsFromRecords(records: List[String]): List[(Future[Unit], CloudStackEvent)] = {
+    records.map(jsonSerializer.deserialize[CloudStackEvent]).collect(handleEvent)
+  }
+
+  @Override
+  def restartEvent(event: CloudStackEvent): (Future[Unit], CloudStackEvent) = {
+    logger.debug(s"restartEvent: $event")
+    handleEvent(event)
+  }
+
+  private val handleEvent = new PartialFunction[CloudStackEvent, (Future[Unit], CloudStackEvent)] {
```

Comment:
It work right, because the "Other" case was writing in the isDefinedAt method, so this method return false for this case. The method "apply" can not to apply for this case.